### PR TITLE
JupyterHub on azure configuration and isntructions

### PR DESCRIPTION
**Pull request by avgils**

### CODE REVIEW

File: deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+
+# JupyterHub on Azure Kubernetes
+
```
Comment (evetion) [inline deploy/README.md:3]: Small abstract please, why and what we get as a result.

Comment (evetion) [reply deploy/README.md:3]: Now the context/overview is a bit unclear, so you could also add a small summary:
- Setup Kubernetes cluster
- Setup Jupyterhub inside the the kubernetes cluster with helm
- Integrate custom container into the above
- Setup authentication on Jupyterhub

File: deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+Login the container registry
+
+	az acr login --name <name container registry>
+
+Attach the container registry to the kubernetes cluster
+
+	az aks update -n <cluster-name> -g <resource group> --attach-acr <name container registry>
+
+The dockerfile for the custom container is stored in thi git repo: Dockerfile-jupyterhub. You first need to download FM from the [build server.](https://build.deltares.nl/buildConfiguration/Dimr_DimrCollector) [This build](https://build.deltares.nl/repository/download/Dimr_DimrCollector/1306704:id/dimrset_lnx64_373.tar.gz) was used for the demo.
```
Comment (evetion) [inline deploy/README.md:42]: thi -> the

File: deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+Follow this documentation to setup a basic JupyterHub installation. Customizations for the PAOTM course will follow in the next sections. The config.yaml in git already contains some settings for the customizations. For this step the basic config.yaml from the documentations can be used.
+
+[Setting up JupyterHub](https://zero-to-jupyterhub.readthedocs.io/en/latest/setup-jupyterhub/setup-jupyterhub.html)
+
+## Create custom container
+For the course we can make a custom container. The container contains the model and notebook for the course.
+First we need a private container registry to store the container. Create a registry on microsoft Azure with the follow commands:
+Resource group should be the same as the kubernetes cluster.
+
+    az acr create --resource-group <resource group> --name <name container registry> --sku Basic
```
Comment (evetion) [inline deploy/README.md:32]: And write down the \<loginServer\> for later use.

File: deploy/README.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+	az acr login --name <name container registry>
+
+Attach the container registry to the kubernetes cluster
+
+	az aks update -n <cluster-name> -g <resource group> --attach-acr <name container registry>
+
+The dockerfile for the custom container is stored in thi git repo: Dockerfile-jupyterhub. You first need to download FM from the [build server.](https://build.deltares.nl/buildConfiguration/Dimr_DimrCollector) [This build](https://build.deltares.nl/repository/download/Dimr_DimrCollector/1306704:id/dimrset_lnx64_373.tar.gz) was used for the demo.
+
+ - Download FM, unzip the file and store it in the models folder.
```
Comment (evetion) [inline deploy/README.md:44]: From where, which version?

File: deploy/README.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+	    guarantee: 1.5
+
+Again update the helm installation:
+	
+	RELEASE=jhub
+	 helm upgrade $RELEASE jupyterhub/jupyterhub --version=0.8.2 -f config.yaml
+
+## Customize storage
+
+By default JupyterHub creates a persistent volume (pv) en persistent volume claim (pvc) for each user. The volume is mounted in /home/jovyan. So all the changes here will be stored in the persistent volume.
```
Comment (evetion) [inline deploy/README.md:88]: jovyan? 🤭

File: deploy/config.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      client_id: ""
+      client_secret: ""
+      token_url: https://${host}/auth/realms/${realm}/protocol/openid-connect/token
+      userdata_url: https://${host}/auth/realms/${realm}/protocol/openid-connect/userinfo
+      userdata_method: GET
+      userdata_params: {'state': 'state'}
+      username_key: email
+  admin:
+    users:
+      - anna.vangils@deltares.nl
```
Comment (evetion) [inline deploy/config.yaml:22]: Maybe add others?

File: deploy/README.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+	      - user10
+	  admin:
+	    users:
+	      - admin
+
+Currently this is replaced with MyDeltares login. Documentation about this can be found [here](https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/authentication.html). The OpenID Connect method can be used to connect with MyDeltares. The following codeblock in the config.yaml is used for the myDeltares connection.
+
+	hub:
+	  extraEnv:
+	    OAUTH2_AUTHORIZE_URL: https://${host}/auth/realms/${realm}/protocol/openid-connect/auth
```
Comment (evetion) [inline deploy/README.md:199]: Is the {host} and {realm} configured somewhere else?

**Review by evetion — COMMENTED** [review]
Nice work!

File: deploy/README.md (hunk 5 above)
Comment (SiggyF) [reply deploy/README.md:88]: That's the default username in jupyter (jovyan == inhabitant of the planet Jupyter)

File: deploy/README.md (hunk 4 above)
Comment (SiggyF) [reply deploy/README.md:44]: build.deltares.nl, build 373  in this example

**Comment by SiggyF** [comment]
I've merged this now. Please incorporate the last comments with fixes from the main branc.

**Pull request merged**